benchmark/framework/cli.py

```diff
@@ -0,0 +1,336 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""Cli implementation for performance tests suits."""
+import ast
+import inspect
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+import click
+from click.core import Argument, Command, Context, Option, Parameter
+
+from matplotlib.axes._axes import Axes  # type: ignore
+
+from .executor import Executor
+from .func_details import BenchmarkFuncDetails
+from .report_printer import PerformanceReport, ReportPrinter
+
+
+class DefaultArgumentsMultiple(Argument):
+    """Multiple arguments with default value."""
+
+    def __init__(self, *args, **kwargs):
+        """Create MultipleArguments instance."""
+        kwargs["nargs"] = -1
+        default = kwargs.pop("default", tuple())
+        super().__init__(*args, **kwargs)
+        self.default = default
+
+    def full_process_value(self, ctx: Context, value: Any) -> Any:
+        """
+        Given a value and context this runs the logic to convert the value as necessary.
+
+        :param ctx: command context
+        :param value: value for option parsed from command line
+
+        :return: value for option
+        """
+        if not value:
+            value = self.default
+        else:
+            value = [self._parse_arg_str(i) for i in value]
+        return super().process_value(ctx, value)
+
+    def _parse_arg_str(self, args: str) -> Tuple[Any]:
+        """
+        Parse arguments string to tuple.
+
+        :param args: arguments string sperated by comma
+
+        :return: tuple of parsed arguments
+        """
+        parsed = ast.literal_eval(args)
+
+        if not isinstance(parsed, tuple):
+            parsed = (parsed,)
+
+        return parsed
+
+
+class TestCli:
+    """Performance test client."""
+
+    def __init__(
+        self,
+        func: Callable,
+        executor_class=Executor,
+        report_printer_class=ReportPrinter,
+    ):
+        r"""
```

Comment:
why `r`?

Reply:
not actual, the reason was inline triple quote comments.
will remove